spec/models/app_identifier_claim_spec.rb

```diff
@@ -0,0 +1,120 @@
+      end
+
+      it "is invalid with no version_number" do
+        claim = described_class.new create_params.merge(version_number: "")
+        expect(claim).to have(1).error_on(:version_number)
+      end
+
+      it "is invalid with no build timestamp" do
+        claim = described_class.new create_params.merge(build_timestamp: "")
+        expect(claim).to have(1).error_on(:build_timestamp)
+      end
+    end
+
+    it "returns true when the IdentifierClaim saves successfully" do
+      app_build = instance_double(App::Build)
+      app_version_build_path = instance_double(AppVersionBuildPath, valid?: true, app_build: app_build)
+      allow(AppVersionBuildPath).to receive(:new).and_return app_version_build_path
+
+      visitor = instance_double(Visitor)
+      identifier = instance_double(Identifier, visitor: visitor)
+      identifier_claim = instance_double(IdentifierClaim, valid?: true, save!: true, identifier: identifier)
+      allow(IdentifierClaim).to receive(:new).and_return identifier_claim
+
+      claim = described_class.new create_params
+
+      expect(claim.save).to eq(true)
+    end
+  end
+
+  describe "#visitor" do
+    it "returns nil before save" do
+      claim = described_class.new create_params
+      expect(claim.visitor).to be_nil
+    end
+
+    it "returns the identifier claim's identifier visitor after save" do
+      app_build = instance_double(App::Build)
+      app_version_build_path = instance_double(AppVersionBuildPath, valid?: true, app_build: app_build)
+      allow(AppVersionBuildPath).to receive(:new).and_return app_version_build_path
+
+      visitor = instance_double(Visitor)
+      identifier = instance_double(Identifier, visitor: visitor)
+      identifier_claim = instance_double(IdentifierClaim, valid?: true, save!: true, identifier: identifier)
+      allow(IdentifierClaim).to receive(:new).and_return identifier_claim
+
+      claim = described_class.new create_params
+      expect(claim.save).to eq(true)
+
+      expect(claim.visitor).to eq(visitor)
+    end
+  end
+
+  describe "#app_build" do
+    it "returns nil before save" do
+      claim = described_class.new create_params
+      expect(claim.app_build).to be_nil
+    end
+
+    it "returns the app version build path's app build after save" do
+      app_build = instance_double(App::Build)
+      app_version_build_path = instance_double(AppVersionBuildPath, valid?: true, app_build: app_build)
+      allow(AppVersionBuildPath).to receive(:new).and_return app_version_build_path
+
+      visitor = instance_double(Visitor)
+      identifier = instance_double(Identifier, visitor: visitor)
+      identifier_claim = instance_double(IdentifierClaim, valid?: true, save!: true, identifier: identifier)
+      allow(IdentifierClaim).to receive(:new).and_return identifier_claim
+
+      claim = described_class.new create_params
+      expect(claim.save).to eq(true)
+
+      expect(claim.app_build).to eq(app_build)
+    end
+  end
```

Comment:
the stubbing in some of these tests is a lil beefy, but the alternative is to use factory bot to set up the db models so i at least wanted to try this out and see if others felt okay about it.